### PR TITLE
Add ability to choose between MM/dd and dd/MM date formats

### PR DESCRIPTION
**Pull request by Verabird**
This set of changes addresses issue #4355, https://github.com/HabitRPG/habitrpg/issues/4355. It adds a "dateFormat" preference to the user model, adds a form on the settings page to choose the format, and changes date displays to use the dateFormat preference. Apologies for the mess of commits, I'm still learning git and went through more than one iteration of how to do this.

### CODE REVIEW

File: public/js/controllers/settingsCtrl.js [hunk 1]
```diff
@@ -65,6 +65,13 @@ habitrpg.controller('SettingsCtrl',
       User.set({'preferences.language': $scope.language.code});
     }
 
+	$scope.availableFormats = ['MM/dd/yyyy','dd/MM/yyyy'];
```
Comment (paglias) [inline public/js/controllers/settingsCtrl.js:68]: ops

Comment (paglias) [inline public/js/controllers/settingsCtrl.js:68]: sorry I missed the function because of how it's written... I'll check it and merge

**Comment by paglias** [comment]
@Verabird I made a change to the indentation and simplified a bit the code but the PR was perfect. Greate work :)
I'll be happy to award you with a badge if you give me your UUID!

**Comment by Verabird** [comment]
uuid: 850cbb7e-4837-41ac-8d32-58d09d97c5bc

**Comment by paglias** [comment]
badge awarded

**Pull request merged**